### ISSUE TITLE
[High] Permanent delete detaches a live Session without stopping its Agent, dropping all later events

### ISSUE THREAD

**Issue by Jstn-1g** [dsh-tauri-desk/dsh-tauri-plugins#22]
## Summary

`dsh-tauri-session` permanent delete can detach a live `Session` from the DSH `SessionStore` without cancelling or disposing its owning `Agent`. The provider request continues, later events are appended only to the detached in-memory object, and persistence/UI/session-event observers receive none of them.

Severity: **High / P1 correctness and data-integrity defect; release blocker for permanent deletion.** Archive/unarchive itself is not broken and can remain enabled.

## Affected versions

- DeepSeek Harness: `dsh-v0.1.2-alpha.1`, `cd5ef8148158c3a752a658978873241fdf8e2bbc`
- DeepSeek Harness Desktop: `f4eb452543a196d9b5400886d327d2d5e217352d`
- `dsh-tauri-session`: `0.5.3`, repository commit `bb7c9f25ad230e648121a28420e71dc11d7d14e3`
- Permanent deletion was merged in https://github.com/dsh-tauri-desk/dsh-tauri-plugins/pull/16

## Exact reproduction

1. Start a DSH session using a gated mock provider so a turn remains active.
2. Archive that session. Official `WorkspaceRegistry.archiveSession()` intentionally accepts a live Session.
3. Invoke the plugin's permanent-delete flow while the provider is still gated.
4. The plugin physically deletes the stored session data, calls the Desktop-injected `SessionStore.remove()`, and then removes workspace/archive accounting.
5. Inspect the exact session and agent registries, then release the provider gate.

Observed against the versions above:

```text
sessions.get(id)                 undefined
agents.get(id)                   same live Agent
provider AbortSignal.aborted     false
session/disposed                 emitted
post-detach session/event count  0
agent completion                 continues through turn/end
```

The detached `Session` object still accepts the later appends in its private in-memory event log, but it no longer publishes them to persistence, UI, or telemetry observers.

Code chain:

- Official archive accepts live Sessions: https://github.com/deepseek-ai/deepseek-harness/blob/cd5ef8148158c3a752a658978873241fdf8e2bbc/packages/workspace/workspace/src/index.ts#L237-L267
- Desktop shim detaches only the Session entry: https://github.com/dsh-tauri-desk/deepseek-harness-desktop/blob/f4eb452543a196d9b5400886d327d2d5e217352d/src-tauri/src/service/patch/session.rs#L12-L18
- Plugin deletion order has no Agent/writer coordination: https://github.com/dsh-tauri-desk/dsh-tauri-plugins/blob/bb7c9f25ad230e648121a28420e71dc11d7d14e3/packages/dsh-tauri-session/src/index.ts#L282-L355

## Expected behavior

Permanent deletion either:

1. rejects `session-busy` before any destructive mutation; or
2. owns an exact Host transaction that cancels and awaits Agent quiescence, retires all persistence writers, deletes/forgets the durable artifact, updates workspace/archive/projection state, and only then publishes success.

After success, both registries must be empty, no later append may occur, and the durable artifact must remain absent after writer retirement.

## Actual behavior and impact

- UI reports removal while Host execution survives invisibly.
- Provider work and possible tool side effects continue.
- Post-detach lifecycle events are silently non-durable.
- Asynchronous persistence retirement can race the physical filesystem deletion.
- Batch/clear-all deletion inherits the same unsafe primitive.

## Smallest safe remediation

1. Immediately disable or fail closed the physical-delete routes while preserving archive/unarchive.
2. Remove the Desktop `SessionStore.remove` injection; `entry.detach()` is not a deletion transaction.
3. Wait for a Host-owned, writer-coordinated delete/forget primitive before re-enabling physical deletion.

Checking only `ctx.agents.get(id)` or `ctx.sessions.get(id)` is useful defense in depth, but it is not a complete fix: it can race a concurrent resume and cannot prove that cold persistence writers are retired.

Development can safely continue with archive/unarchive and session browsing enabled. Permanent deletion should remain disabled until the lifecycle transaction exists.

Related upstream API gap: https://github.com/deepseek-ai/deepseek-harness/discussions/5057
